README.md

```diff
@@ -1,6 +1,233 @@
+  "data": {
+    "function": "get_widget_data",
+    "input_arguments": {
+      "widget_uuid": "c276369e-e469-4689-b5fe-3f8c76f7c45a"
+    }
+  }
+}
+```
+
+### Handling Function Calls in Your Copilot
+
+Once a function call is required, the copilot should send a copilotFunctionCall event. Terminal Pro will execute the specified function and return the result. You must handle these function calls correctly to continue the conversation.
+
+For example, if your copilot requests widget data:
+
+- Emit the copilotFunctionCall SSE.
+- Wait for Terminal Pro to provide the result.
+- Include the result in subsequent responses to complete the flow.
+
+**Note:** The copilot must also include the function call response in any further requests to maintain context.
+
+## Understanding the Copilot Configuration (copilots.json)
+
+To integrate your custom copilot with Terminal Pro, you need to configure the copilots.json file. This file defines how your custom copilot connects with Terminal Pro, including which features it supports and where requests should be sent.
+
+Here's an example copilots.json configuration:
+
+```json
+{
+  "example_copilot": {
+    "name": "Mistral Example Co. Copilot",
+    "description": "AI-powered financial copilot that uses Mistral Large as its LLM.",
+    "image": "https://github.com/OpenBB-finance/copilot-for-terminal-pro/assets/14093308/5c7518ad-3997-4fd7-9549-909eddd272c4",
+    "hasStreaming": true,
+    "hasFunctionCalling": true,
+    "hasDocuments": false,
+    "endpoints": {
+      "query": "http://localhost:7777/v1/query"
+    }
+  }
+}
+```
+
+- `example_copilot`: This is the unique identifier for your copilot.
+- `name`: The display name of your copilot. This name is used within the Terminal Pro UI to allow users to identify and select the copilot.
+- `description`: A short description of your copilot.
+- `image`: A URL to an image representing your copilot. This image will be displayed in the Terminal Pro interface.
+- `hasStreaming`: A boolean value indicating if the copilot supports streaming responses via SSEs. If set to `true`, Terminal Pro will expect the copilot to return responses in real-time chunks.
+- `hasFunctionCalling`: A boolean value indicating if the copilot supports function calling. If set to `true`, Terminal Pro will be able to make specific function calls such as retrieving widget data or accessing client-side resources.
+- `hasDocuments`: A boolean value indicating if the copilot supports processing documents, such as user-uploaded files. If set to `false`, document processing features will not be available for this copilot.
+- `endpoints`: A set of URLs that Terminal Pro will use to interact with your copilot. In this example:
+  - `query`: The URL for the main endpoint where Terminal Pro sends user queries (POST requests). This endpoint is where your copilot listens for incoming messages and responds accordingly.
+
+## Example Scenarios and Use Cases
+
+### Requesting Widget Data
```

Comment:
Start with the simplest case! Just a regular conversation with Copilot on an empty dashboard.